lgc/include/lgc/patch/PatchPreparePipelineAbi.h

```diff
@@ -0,0 +1,98 @@
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+***********************************************************************************************************************
+* @file  PatchPrepareAbi.cpp
+* @brief LLPC source file: contains declaration of class lgc::PatchPreparePipelineAbi.
+***********************************************************************************************************************
+*/
+
+#include "lgc/patch/Patch.h"
+#include "lgc/state/PipelineShaders.h"
+#include "lgc/state/PipelineState.h"
+#include "lgc/state/TargetInfo.h"
+
+namespace lgc {
+
+// =====================================================================================================================
+// Pass to prepare the pipeline ABI
+class PatchPreparePipelineAbi final : public Patch, public llvm::PassInfoMixin<PatchPreparePipelineAbi> {
+public:
+  PatchPreparePipelineAbi(bool onlySetCallingConvs = false);
+
+  llvm::PreservedAnalyses run(llvm::Module &module, llvm::ModuleAnalysisManager &analysisManager);
+
+  bool runImpl(llvm::Module &module, PipelineShadersResult &pipelineShaders, PipelineState *pipelineState);
+
+  static llvm::StringRef name() { return "Patch LLVM for preparing pipeline ABI"; }
+
+private:
+  void setCallingConvs(llvm::Module &module);
+
+  void setRemainingCallingConvs(llvm::Module &module);
+
+  void mergeShaderAndSetCallingConvs(llvm::Module &module);
+
+  void setCallingConv(ShaderStage stage, llvm::CallingConv::ID callingConv);
+
+  void setAbiEntryNames(llvm::Module &module);
+
+  void addAbiMetadata(llvm::Module &module);
+
+  PipelineState *m_pipelineState;           // Pipeline state
+  PipelineShadersResult *m_pipelineShaders; // API shaders in the pipeline
+
+  bool m_hasVs;  // Whether the pipeline has vertex shader
+  bool m_hasTcs; // Whether the pipeline has tessellation control shader
+  bool m_hasTes; // Whether the pipeline has tessellation evaluation shader
+  bool m_hasGs;  // Whether the pipeline has geometry shader
+
+  GfxIpVersion m_gfxIp; // Graphics IP version info
+
+  const bool m_onlySetCallingConvs; // Whether to only set the calling conventions
+};
+
+// =====================================================================================================================
+// Pass to prepare the pipeline ABI
+class LegacyPatchPreparePipelineAbi final : public llvm::ModulePass {
+public:
+  static char ID;
```

Comment:
I don't know if there's a standard for that, but I suggest you move this static member to the bottom of the `public` part so this is consistent with the other class declarations.

Reply:
This is also unrelated to this patch, and that code will also eventually be removed once the switch to the new pass manager is completed.